src/services/api.ts

```diff
@@ -43,11 +45,26 @@ export class APIService {
 
   setAuth(token?: string) {
     this.token = token;
-    const auth = token
-      ? {
-          headers: { Authorization: 'Bearer ' + token },
-        }
-      : {};
+
+    const auth: Record<string, unknown> = {};
+    if (token) {
+      const authHeader = 'Bearer ' + token;
+      auth.headers = { Authorization: authHeader };
+
+      // add interceptor so GQty requests have proper authorization
+      this.interceptor = hasuraAxios.interceptors.request.use(
```

Comment:
Another thing that interceptors can be used for is triggering re-authorization when a token expires